src/main/java/edu/ie3/datamodel/io/source/csv/CsvRawGridSource.java

```diff
@@ -0,0 +1,471 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.source.csv;
+
+import edu.ie3.datamodel.io.FileNamingStrategy;
+import edu.ie3.datamodel.io.factory.EntityFactory;
+import edu.ie3.datamodel.io.factory.input.*;
+import edu.ie3.datamodel.io.source.RawGridSource;
+import edu.ie3.datamodel.io.source.TypeSource;
+import edu.ie3.datamodel.models.UniqueEntity;
+import edu.ie3.datamodel.models.input.*;
+import edu.ie3.datamodel.models.input.connector.LineInput;
+import edu.ie3.datamodel.models.input.connector.SwitchInput;
+import edu.ie3.datamodel.models.input.connector.Transformer2WInput;
+import edu.ie3.datamodel.models.input.connector.Transformer3WInput;
+import edu.ie3.datamodel.models.input.connector.type.LineTypeInput;
+import edu.ie3.datamodel.models.input.connector.type.Transformer2WTypeInput;
+import edu.ie3.datamodel.models.input.connector.type.Transformer3WTypeInput;
+import edu.ie3.datamodel.models.input.container.RawGridElements;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * //ToDo: Class Description Nothing is buffered -> for performance one might consider reading
+ * nodes, operators etc. first and then passing in all required collections, otherwise reading is
+ * done in a hierarchical cascading way to get all elements needed TODO description needs hint that
+ * Set does NOT mean uuid uniqueness
+ *
+ * <p>// todo performance improvements in all sources to make as as less possible recursive stream
+ * calls on files
```

Comment:
Is this something that has to be addressed now? Or is it a good idea to create a new issue?

Reply:
This is an idea for the future and has been removed now, as we don't use `Future`s atm for reading. Would have been possible (and still is) but from a performance perspective there is no need to address this _yet_